### ISSUE TITLE
Speed shoes themes broken on latest test release

### ISSUE THREAD

**Issue by TimerBunneh** [ShadowXeldron/ShadowXeldrons-S3AIR-Mods#2]
Turns out the speed shoes themes are broken on the latest test release. How so? They don't seem to play at all. In fact, the music doesn't change one bit--at least with my mod enabled.

My guess is because Standalone\audio.lemon got an update, and that got implemented into my latest mod, which made your speed shoes unhappy. The latest test release (as of this typing) is 23.05.14.1.

Take this (your speed shoes script should also begin with #if STANDALONE):
```
function void Standalone.setFastMusicFlag(u8 flag, bool enable)
{
	// Update flags
	bool wasFastMusicMode = (fastMusicFlags != 0)
	if (enable)
		fastMusicFlags |= flag
	else
		fastMusicFlags &= ~flag

	// Any change?
	bool shouldBeFastMusicMode = (fastMusicFlags != 0)
	if (shouldBeFastMusicMode != wasFastMusicMode)
	{
		// Switch to fast music, or back to normal
		//  -> In the Oxygen engine, this is implemented as an audio modifier for channel 0
		if (shouldBeFastMusicMode)
		{
			Audio.enableAudioModifier(0, 0, "_fast", 1.25f)
		}
		else
		{
			Audio.disableAudioModifier(0, 0)
		}
	}
}

function void Standalone.setFastMusicFlagWithSpeedup(u8 flag, float relativeSpeed)
{
	if (relativeSpeed <= 1.0f)
	{
		Standalone.setFastMusicFlag(flag, false)
	}
	else
	{
		fastMusicFlags |= flag
		Audio.enableAudioModifier(0, 0, "_fast", relativeSpeed)
	}
}
#endif
```

**Comment by ShadowXeldron** [comment]
I assume this would be about right?

```
#if STANDALONE

function void Character.updateBonusEffects() 
{
	
	bool speedupped = (char.bonus_effect & char.bonus.SPEED_UP) 
	base.Character.updateBonusEffects()
	if (speedupped)
	{
		playSpeedSong()
	}
	if ((speedupped) && (((char.bonus_effect & char.bonus.SPEED_UP)) == 0))
	{	
		chooseFittingMusic()
	}
}

function void Standalone.setFastMusicFlag(u8 flag, bool enable)
{
	// Update flags
	bool wasFastMusicMode = (fastMusicFlags != 0)
	if (enable)
		fastMusicFlags |= flag
	else
		fastMusicFlags &= ~flag

	// Any change?
	bool shouldBeFastMusicMode = (fastMusicFlags != 0)
	if (shouldBeFastMusicMode != wasFastMusicMode)
	{
		// Switch to fast music, or back to normal
		//  -> In the Oxygen engine, this is implemented as an audio modifier for channel 0
		if (shouldBeFastMusicMode)
		{
			Audio.enableAudioModifier(0, 0, "_fast", 1.25f)
		}
		else
		{
			Audio.disableAudioModifier(0, 0)
		}
	}
}

function void Standalone.setFastMusicFlagWithSpeedup(u8 flag, float relativeSpeed)
{
	if (relativeSpeed <= 1.0f)
	{
		Standalone.setFastMusicFlag(flag, false)
	}
	else
	{
		fastMusicFlags |= flag
		Audio.enableAudioModifier(0, 0, "_fast", relativeSpeed)
	}
}
#endif
```

**Comment by ShadowXeldron** [comment]
Merged upstream, but with the following changes:
 - The function "Standalone.setFastMusicFlagWithSpeedup" has been commented out because it seems to be identical to the basegame a
 - All floats have been removed for compatability with the current stable version. It does not work on Android and requires manual compilation on Linux, so all floats have been ommited until ther previews are officially available on those platforms.

I tested the script on the stable version and a manually compiled copy of the preview. I'll leave this issue open until I have confirmation that it works on the stable version across all platforms.
